packages/bpk-component-infinite-scroll/src/BpkInfiniteScroll.js

```diff
@@ -0,0 +1,172 @@
+      this.onScroll = props.onScroll;
+      this.handleIntersection = this.handleIntersection.bind(this);
+      this.handleSeeMoreClick = this.handleSeeMoreClick.bind(this);
+      this.handleKeyPress = this.handleKeyPress.bind(this);
+      // TODO add polyfill for IE
+      this.observer = new IntersectionObserver(
+        (...args) => this.handleIntersection(...args),
+        { threshold: 1 },
+      );
+    }
+
+    componentDidMount() {
+      if (this.sentinel) {
+        this.observer.observe(this.sentinel);
+      }
+    }
+
+    componentDidUpdate() {
+      if (this.sentinel && this.state.index > 0) {
+        this.observer.observe(this.sentinel);
+      }
+    }
+
+    async loadMore() {
+      const { index, elementsToRender } = this.state;
+      const { elementsPerScroll, seeMoreAfter } = this.props;
+      const nextElements = await this.onRequestElements(
+        index,
+        elementsPerScroll,
+      );
+      if (nextElements && nextElements.length > 0) {
+        const nextIndex = index + elementsPerScroll;
+        return {
+          index: nextIndex,
+          elementsToRender: elementsToRender.concat(nextElements),
+          showSeeMore: seeMoreAfter === index / elementsPerScroll,
+        };
+      }
+      this.props.onScrollFinished({
+        totalNumberElements: elementsToRender.length,
+      });
+      return {
+        isListFinished: true,
+      };
+    }
+
+    handleIntersection(entries) {
+      entries.forEach(async entry => {
+        if (entry.intersectionRatio === 1) {
+          this.props.onScroll({ currentIndex: this.state.index });
+          this.observer.unobserve(this.sentinel);
+          const newState = await this.loadMore();
+          this.setState(newState);
+        }
+      });
+    }
+
+    async handleSeeMoreClick(e) {
+      this.props.onClickSeeMore(e);
+      this.setState(await this.loadMore());
+    }
+
+    handleKeyPress(e) {
+      const { key } = e;
+      if (key === 'Enter' || key === ' ') {
+        this.handleSeeMoreClick(e);
+      }
+    }
+
+    render() {
+      const { elementsToRender, isListFinished, showSeeMore } = this.state;
+      const { seeMoreComponent } = this.props;
+
+      let spinnerOrButton = null;
+
+      if (!isListFinished) {
+        if (showSeeMore) {
+          spinnerOrButton = (
+            <div
+              role="button"
+              tabIndex={0}
+              onKeyPress={this.handleKeyPress}
+              onClick={this.handleSeeMoreClick}
+            >
+              {seeMoreComponent}
+            </div>
+          );
+        } else {
+          spinnerOrButton = (
+            <div
+              ref={spinner => {
+                this.sentinel = spinner;
+              }}
+              className={STYLES['bpk-sentinel']}
+            >
+              <BpkSpinner type={SPINNER_TYPES.primary} />
+            </div>
+          );
+        }
+      }
+
+      return (
+        <div className={STYLES['bpk-infiniteList']}>
```

Comment:
We should spread arbitrary props here i.e. `{...rest}`

Reply:
I'm already spreading the props to the `ComponentToExtend` in the line below that one. That can create some problem for example when passing `className` 🤔